test/e2e/e2e_test.go

```diff
@@ -201,6 +202,59 @@ var _ = ginkgo.Describe("leaderWorkerSet e2e tests", func() {
 		}
 	})
 
+	ginkgo.It("When changing subdomainPolicy, adds correct env vars", func() {
+		leaderPodSpec := testing.MakeLeaderPodSpecWithTPUResource()
+		workerPodSpec := testing.MakeWorkerPodSpecWithTPUResource()
+		lws := testing.BuildLeaderWorkerSet(ns.Name).Replica(1).Size(2).LeaderTemplateSpec(leaderPodSpec).WorkerTemplateSpec(workerPodSpec).Obj()
+		testing.MustCreateLws(ctx, k8sClient, lws)
+		testing.ExpectValidPods(ctx, k8sClient, lws, &corev1.PodList{})
+		testing.UpdateSubdomainPolicy(ctx, k8sClient, lws, leaderworkerset.SubdomainUniquePerReplica)
+		lwsPods := &corev1.PodList{}
+		testing.ExpectValidPods(ctx, k8sClient, lws, lwsPods)
+
+		for _, pod := range lwsPods.Items {
+			gomega.Expect(testing.HasTPUEnvVarsPopulated(pod)).To(gomega.BeTrue())
+			if pod.Labels[leaderworkerset.GroupIndexLabelKey] == "0" {
```

Comment:
The GroupIndex refers to replica index, not the pod index within a replica. Since you have one replica, this will be 0 for both pods, and so I am confused how come the test is passing because the check in the else should fail, what am I missing?

Reply:
It never reaches the else statement, so it never fails that portion of the test. I'll update the number of replicas to two to verify that the subdomain name is different on both replicas